biochatter/api_agent.py

```diff
@@ -0,0 +1,441 @@
+                print(f"{question_uuid} results not ready, waiting...")
+                time.sleep(15)
+            elif "Status=FAILED" in status_text:
+                with open(f"{save_path}{file_name}", "w") as file:
+                    file.write("BLAST query FAILED.")
+            elif "Status=UNKNOWN" in status_text:
+                with open(f"{save_path}{file_name}", "w") as file:
+                    file.write("BLAST query expired or does not exist.")
+                raise
+            elif "Status=READY" in status_text:
+                if "ThereAreHits=yes" in status_text:
+                    print(
+                        f"{question_uuid} results are ready, retrieving and saving..."
+                    )
+                    results_response = requests.get(
+                        base_url, params=get_results_params
+                    )
+                    results_response.raise_for_status()
+                    # Save the results to a file
+                    print(f"{save_path}{file_name}")
+                    with open(f"{save_path}{file_name}", "w") as file:
+                        file.write(results_response.text)
+                    print(f"Results saved in BLAST_results_{question_uuid}.txt")
+                    break
+                else:
+                    with open(f"{save_path}{file_name}", "w") as file:
+                        file.write("No hits found")
+                    break
+            else:
+                print("Unknown status")
+                with open(f"{save_path}{file_name}", "w") as file:
+                    file.write("Unknown status")
+                break
+        if attempt == max_attempts - 1:
+            raise TimeoutError(
+                "Maximum attempts reached. Results may not be ready."
+            )
+        return file_name
+
+    def read_first_n_lines(self, file_path: str, n: int):
+        """
+        Reads the first n lines from a file and returns them as a string.
+
+        Args:
+            file_path (str): The path to the file.
+            n (int): The number of lines to read.
+
+        Returns:
+            str: The first n lines from the file as a string.
+
+        Raises:
+            FileNotFoundError: If the file is not found.
+            Exception: If any other error occurs during file reading.
+
+        """
+        try:
+            with open(file_path, "r") as file:
+                lines = []
+                for i in range(n):
+                    line = file.readline()
+                    if not line:
+                        break
+                    lines.append(line.strip())
+                # to test:
+                # more efficient with \n or without?
+                return "\n".join(lines)
+        except FileNotFoundError:
+            return "The file was not found."
+        except Exception as e:
+            return f"An error occurred: {e}"
+
+    def answer_extraction(self, question: str, file_path: str, n: int) -> str:
```

Comment:
would rather put this into a third class, the Interpreter (in addition to the generic QueryBuilder and Fetcher classes)